app/app.config.js

```diff
@@ -69,12 +69,18 @@
       )
 
       .run(['$rootScope', '$location', 'AuthService', function($rootScope, $location, AuthService) {
-            $rootScope.$on('$locationChangeStart', function () {
+            $rootScope.$on('$locationChangeStart', function (event, next, current) {
                 // redirect to login page if not logged in and trying to access a restricted page
+                // prevent access to login page when user already logged in and authenticated
                 var restrictedPage = $.inArray($location.path(), ['/login']) === -1;
                 var loggedIn = AuthService.isAuthenticated();
-                if (restrictedPage && !loggedIn) {
-                      $location.path('/login');
+                if (restrictedPage) {
+                    if(!loggedIn) {
```

Comment:
Why do you need a nested if statement if you are not doing anything else?

Reply:
The following code is equivalent to this:
if (restrictedPage && !loggedIn) {
    $location.path('/login');
}
else if(loggedIn && !restrictedPage) {
    event.preventDefault();
}
If the page is restricted and the user is not logged in he would be redirected to login page.
If the user is logged in and the page is not restricted (login page) then he should not be able to access it as he is already authenticated and logged in.
And a logged in user can access the restricted pages.

So to avoid two condition checks in the else if I thought it would be better to use a nested statement.

In case the reason does not seem correct please let me know, I will make the required changes. @botraunak

Reply:
The following code is equivalent to this:
if (restrictedPage && !loggedIn) {
$location.path('/login');
}
else if(loggedIn && !restrictedPage) {
event.preventDefault();
}
If the page is restricted and the user is not logged in he would be redirected to login page.
If the user is logged in and the page is not restricted (login page) then he should not be able to access it as he is already authenticated and logged in.
And a logged in user can access the restricted pages.

So to avoid two condition checks in the else if I thought it would be better to use a nested statement.

In case the reason does not seem correct please let me know, I will make the required changes. @botraunak